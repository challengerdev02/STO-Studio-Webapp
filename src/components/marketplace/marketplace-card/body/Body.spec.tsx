import {
  cleanup,
  queryByRole,
  queryByText,
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BodyComponent } from './index';

describe('Container', function () {
  afterEach(cleanup);
  let props = {
    title: 'Main Title',
    placeBid: jest.fn(),
    itemPrice: 'Coin Value',
    rates: '2.0',
    iconURL: 'www.url.com',
    count: '10',
    inWishlist: true,
    urlList: [],
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<BodyComponent {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have Content "This is a test for Content"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByText(document.body, 'Main Title')).toBeInTheDocument();
  });
  it('should have Wishlist', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByRole(document.body, 'InWishlist')).toBeInTheDocument();
    expect(queryByRole(document.body, 'NotInWishlist')).not.toBeInTheDocument();
  });

  it('should Not have Wishlist', () => {
    // @ts-ignore
    const { container } = setup({ inWishlist: false });
    expect(queryByRole(document.body, 'InWishlist')).not.toBeInTheDocument();
    expect(queryByRole(document.body, 'NotInWishlist')).toBeInTheDocument();
  });
});
