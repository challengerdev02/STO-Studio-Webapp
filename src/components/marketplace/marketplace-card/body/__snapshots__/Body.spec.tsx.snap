// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match with snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL dIxYVf"
  >
    <span
      class="ant-typography sc-gKclnd elkVOZ"
      font-size="14px"
      font-weight="500"
    >
      Main Title
    </span>
    <div
      class="sc-dkPtRN sc-hKwDye kToXgV rBElr"
      height="24px"
      width="24px"
    >
      <img
        alt="Main Title"
        class="sc-iCfMLu bpDNvp"
        src="www.url.com"
        width="100%"
      />
    </div>
  </div>
  <div
    class="sc-bdvvtL dIxYVf"
  >
    <span>
      <span
        class="ant-typography sc-gKclnd fILiiP"
        font-size="12px"
        font-weight="600"
        line_height="20px"
      >
        Coin Value
      </span>
      <span
        class="ant-typography sc-gKclnd KfwZx"
        font-size="12px"
        font-weight="500"
        margin="0 0 0 4.5px"
      >
        2.0
      </span>
    </span>
    <span>
      <span
        aria-label="heart"
        class="anticon anticon-heart"
        role="InWishlist"
      >
        <svg
          aria-hidden="true"
          data-icon="heart"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
          />
        </svg>
      </span>
      <span
        class="ant-typography sc-gKclnd foTPFP"
        font-size="14px"
        font-weight="600"
        line_height="160%"
        margin=" 0 0 0 4.5px"
      >
        10
      </span>
    </span>
  </div>
  <div
    class="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    class="sc-bdvvtL dIxYVf"
  >
    <div
      class="sc-bdvvtL sc-jRQBWg cxIwvg gqzvZh"
      width="auto"
    />
    <span>
      <button
        class="ant-btn ant-btn-link"
        style="padding: 0px; font-size: 14px; font-weight: 600; margin: 0px 0px 0px 4.5px; line-height: 160%;"
        type="button"
      >
        <span
          aria-label="sliders"
          class="anticon anticon-sliders"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="sliders"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M320 224h-66v-56c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v56h-66c-4.4 0-8 3.6-8 8v560c0 4.4 3.6 8 8 8h66v56c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8v-56h66c4.4 0 8-3.6 8-8V232c0-4.4-3.6-8-8-8zm-60 508h-80V292h80v440zm644-436h-66v-96c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v96h-66c-4.4 0-8 3.6-8 8v416c0 4.4 3.6 8 8 8h66v96c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8v-96h66c4.4 0 8-3.6 8-8V304c0-4.4-3.6-8-8-8zm-60 364h-80V364h80v296zM612 404h-66V232c0-4.4-3.6-8-8-8h-52c-4.4 0-8 3.6-8 8v172h-66c-4.4 0-8 3.6-8 8v200c0 4.4 3.6 8 8 8h66v172c0 4.4 3.6 8 8 8h52c4.4 0 8-3.6 8-8V620h66c4.4 0 8-3.6 8-8V412c0-4.4-3.6-8-8-8zm-60 145a3 3 0 01-3 3h-74a3 3 0 01-3-3v-74a3 3 0 013-3h74a3 3 0 013 3v74z"
            />
          </svg>
        </span>
        <span>
          Place a bid
        </span>
      </button>
    </span>
  </div>
</div>
`;
